# **Import libraries**

In [65]:
import pandas as pd
import requests
from datetime import datetime
from sklearn.linear_model import LinearRegression

In [66]:
#Get data from Open Meteo
url = "https://api.open-meteo.com/v1/forecast?latitude=53.35&longitude=-6.26&daily=temperature_2m_max&past_days=90&forecast_days=7&timezone=auto"

data = requests.get(url).json()

print(data)

{'latitude': 53.34692, 'longitude': -6.2677307, 'generationtime_ms': 0.6549358367919922, 'utc_offset_seconds': 0, 'timezone': 'Europe/Dublin', 'timezone_abbreviation': 'GMT', 'elevation': 11.0, 'daily_units': {'time': 'iso8601', 'temperature_2m_max': '°C'}, 'daily': {'time': ['2025-12-15', '2025-12-16', '2025-12-17', '2025-12-18', '2025-12-19', '2025-12-20', '2025-12-21', '2025-12-22', '2025-12-23', '2025-12-24', '2025-12-25', '2025-12-26', '2025-12-27', '2025-12-28', '2025-12-29', '2025-12-30', '2025-12-31', '2026-01-01', '2026-01-02', '2026-01-03', '2026-01-04', '2026-01-05', '2026-01-06', '2026-01-07', '2026-01-08', '2026-01-09', '2026-01-10', '2026-01-11', '2026-01-12', '2026-01-13', '2026-01-14', '2026-01-15', '2026-01-16', '2026-01-17', '2026-01-18', '2026-01-19', '2026-01-20', '2026-01-21', '2026-01-22', '2026-01-23', '2026-01-24', '2026-01-25', '2026-01-26', '2026-01-27', '2026-01-28', '2026-01-29', '2026-01-30', '2026-01-31', '2026-02-01', '2026-02-02', '2026-02-03', '2026-02-

In [50]:
#variable
dates = data["daily"]["time"]
temps = data["daily"]["temperature_2m_max"]

In [51]:
#DataFrame
df = pd.DataFrame({
    "date": dates,
    "temperature": temps
})

In [52]:
#Convert date column
df["date"] = pd.to_datetime(df["date"])

In [53]:
# Convert date to ordinal
df["date_num"] = df["date"].map(datetime.toordinal)

In [54]:
# Trained data for the past 90 days
train = df.iloc[:90]

In [55]:
# Drop rows with NaN values
train = train.dropna(subset=["temperature"])

In [56]:
# Features and target
X = train[["date_num"]]
y = train["temperature"]

In [57]:
# Train model
model = LinearRegression()
model.fit(X, y)

LinearRegression()

In [58]:
# User input
predict_date = input("Prediction Date: (YYYY-MM-DD): ")

Prediction Date: (YYYY-MM-DD): 2026-03-19


# **Convert input date**

In [62]:
predict_date_dt = pd.to_datetime(predict_date)
predict_date_num = predict_date_dt.toordinal()

# **Predicted Max Temperature**

In [63]:
prediction = model.predict([[predict_date_num]])

print("Predicted Max Temperature:", prediction[0])

Predicted Max Temperature: 12.5437769961718


/usr/local/lib/python3.12/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# **Actual Forecast Temperature**

In [64]:
future_df = df.iloc[90:]

actual = future_df.loc[future_df["date"] == predict_date_dt, "temperature"].values

print("Actual Forecast Temperature from Open Meteo:", actual)

Actual Forecast Temperature from Open Meteo: [10.]


## **The machine learning model predicted a maximum temperature of 12.54°C for 2026-03-19. According to the Open-Meteo data, the maximum temperature recorded for that day is 10.0°C.**